modules/mongodb/mongodb.go

```diff
@@ -73,6 +73,31 @@ func WithPassword(password string) testcontainers.CustomizeRequestOption {
 	}
 }
 
+// WithReplicaSet configures MongoDB to start with a replica set named "rs".
+func WithReplicaSet() testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		req.Cmd = append(req.Cmd, "--replSet", "rs")
+		req.LifecycleHooks = append(req.LifecycleHooks, testcontainers.ContainerLifecycleHooks{
+			PostReadies: []testcontainers.ContainerHook{
+				func(ctx context.Context, c testcontainers.Container) error {
+					cIP, err := c.ContainerIP(ctx)
+					if err != nil {
+						return err
+					}
+
+					cmd := eval("rs.initiate({ _id: 'rs', members: [ { _id: 0, host: '%s:27017' } ] })", cIP)
+
+					if exitCode, _, err := c.Exec(ctx, cmd); err != nil || exitCode != 0 {
+						return fmt.Errorf("failed to initiate the replica set with status %d: %s", exitCode, err)
+					}
+
+					return nil
+				},
+			},
+		})
+	}
+}
```

Comment:
Could you convert this postReadies hook into a wait strategy that is added to the existing one? The wait strategies run and after that the container is marked as ready. So running the replica set as a wait strategy will define a readiness check in the case the functional opt is passed. Wdyt?

I can help you out if needed.

Reply:
Hi @mdelapenya, I've been working on converting to a wait strategy. However, the members need to use either `localhost:[container-ip]` or `[container-name]:27017`, and I'm unsure how to obtain the container's name or IP using a `GenericContainerRequest`. Do you have any suggestions?  https://github.com/testcontainers/testcontainers-go/pull/2469/commits/504cd35666937da3063f42da5d9b506ba7a955ee#diff-ad47d995563c853ad9c6745c55e4ffda6237cb13df6917c8f76bc2ad3c78609aR82-R92

Reply:
Oh you're totally right: because the wait strategy must be built before the container is created, we do not have access to the container IP. I think you can use this very same approach: https://github.com/testcontainers/testcontainers-go/blob/5fa65489ac33e7a6a15e4599739b2c717cf3a1a9/modules/kafka/kafka.go#L84-L87

- create an adhoc wait strategy in the post starts of the container, and there you have access to the container instance.